### PR TITLE
Allow to register multiple Indico users at once

### PR DESCRIPTION
**Pull request by florv**
For now, the registrations are created with the personal data only and other fields are left empty.

### CODE REVIEW

File: indico/htdocs/js/indico/modules/registration/reglists.js [hunk 1]
```diff
@@ -191,6 +191,24 @@
         $('.js-add-user').on('click', function() {
             principal.principalfield('choose');
         });
+
+        $('.js-add-multiple-users').on('click', function(e) {
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/reglists.js]: You can probably use `$('.js-add-multiple-users').ajaxDialog({...})` here

File: indico/modules/events/registration/blueprint.py [hunk 2]
```diff
@@ -114,6 +115,8 @@
                  'generate_static_url', RHRegistrationListStaticURL, methods=('POST',))
 _bp.add_url_rule('/manage/registration/<int:reg_form_id>/registrations/create', 'create_registration',
                  RHRegistrationCreate, methods=('GET', 'POST'))
+_bp.add_url_rule('/manage/registration/<int:reg_form_id>/registrations/create_multiple',
```
Comment (ThiefMaster) [inline indico/modules/events/registration/blueprint.py]: `s/_/-/`

File: indico/modules/events/registration/templates/management/registration_create_multiple.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{% from 'forms/_form.html' import form_header, form_footer, form_rows %}
+{% from 'message_box.html' import message_box %}
+
+
+{{ form_header(form, id='') }}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/management/registration_create_multiple.html]: just omit the attribute if you don't want an ID

File: indico/web/forms/fields.py [hunk 4]
```diff
@@ -261,12 +261,18 @@ class PrincipalListField(HiddenField):
 
     widget = JinjaWidget('forms/principal_list_widget.html', single_kwargs=True)
 
+    def __call__(self, *args, **kwargs):
+        kwargs['open_immediately'] = self.open_immediately
```
Comment (ThiefMaster) [inline indico/web/forms/fields.py]: why not just access `field.open_immediately` in the template?

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 5]
```diff
@@ -521,6 +521,29 @@ def _process(self):
...
+class RHRegistrationCreateMultiple(RHManageRegFormBase):
+    """Create multiple registrations for Indico users (management area)"""
+
+    def _register_user_personal_data_only(self, user, notify):
+        data = {pdt.name: getattr(user, pdt.name, None) for pdt in PersonalDataType}
+        with db.session.no_autoflush:
+            create_registration(self.regform, data, management=True, notify_user=notify)
+
+    def _process(self):
+        form = CreateMultipleRegistrationsForm(regform=self.regform, open_add_user_dialog=request.method == 'GET')
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py]: - get the default value for `notify_user` from the session (and store it on successful submission)
- I'd wrap `request.method == 'GET'` in parentheses. looks pretty awful without them IMO

File: indico/modules/events/registration/templates/management/registration_create_multiple.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+{{ form_rows(form) }}
+
+{{ message_box('info', message=_('The registrations will be filled with personnal data only. Additional fields will be left blank.')) }}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/management/registration_create_multiple.html]: `personal`

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 7]
```diff
@@ -521,6 +521,31 @@ def _process(self):
                                                     user_data=self._get_user_data(), management=True)
 
 
+class RHRegistrationCreateMultiple(RHManageRegFormBase):
+    """Create multiple registrations for Indico users (management area)"""
+
+    def _register_user_personal_data_only(self, user, notify):
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py]: isn't the `_personal_data_only` a bit verbose? ;)

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 8]
```diff
@@ -521,6 +521,31 @@ def _process(self):
...
+
+    def _process(self):
+        form = CreateMultipleRegistrationsForm(regform=self.regform, open_add_user_dialog=(request.method == 'GET'),
+                                               notify_users=session.get('registration_notify_user_default', True))
+
+        if form.validate_on_submit():
+            session['registration_notify_user_default'] = form.notify_users.data
+            for user in form.user_principals.data:
+                self._register_user_personal_data_only(user, form.notify_users.data)
+            reg_list_config = _get_reg_list_config(regform=self.regform)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py]: is this code block (and the lines below) repeated, e.g. in the other RH to add a registration? if yes it'd be useful to move it into a function. e.g. `jsonify_registration_list(regform)`

File: indico/modules/events/registration/forms.py [hunk 9]
```diff
@@ -299,3 +299,23 @@ def validate_json(self, field):
...
+    user_principals = PrincipalListField(_("Indico users"), [DataRequired()])
+    notify_users = BooleanField(_("Send e-mail notifications"),
+                                default=True,
+                                description=_("Notify the users about the registration."),
+                                widget=SwitchWidget())
+
+    def __init__(self, *args, **kwargs):
+        self._regform = kwargs.pop('regform')
+        super(CreateMultipleRegistrationsForm, self).__init__(*args, **kwargs)
+        self.user_principals.open_immediately = kwargs.pop('open_add_user_dialog', False)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/forms.py]: I'd pop before the super call

File: indico/modules/events/registration/controllers/management/reglists.py (hunk 7 above)
Comment (florv) [inline indico/modules/events/registration/controllers/management/reglists.py]: Well, I could shorten it to `_register_user` and add a comment stating that it fills only the personal data fields.

**Comment by ThiefMaster** [comment]
Superseded by #2556

**Pull request closed without merging**